### ISSUE TITLE
How to get function return from xmp.spawn distributed processes

### ISSUE THREAD

**Issue by 8key** [pytorch/xla#2613]
Wonder how do we get value returned from spawned functions.
For example, if accuracy is calculated in each core, and i want it to be returned to the main function

```
def _mp_fn():
    #some training and valuation code here

    return accuracy
```

```
accuracy = xmp.spawn(_mp_fn, nprocs=8)
```

In multiprocessor library we can do something like

```
if __name__ == '__main__':
    p = Pool(processes=20)
    data = p.map(job, [i for i in range(20)])
    p.close()
    print(data)
```

How do we do it with xmp.spawn?

**Comment by taylanbil** [comment]
I'd achieve this w/o using the return value (which does not exist), but doing this:

```python
def _mp_fn():
    #some training and valuation code here

    #return accuracy
    accuracy = accuracy.item()  # if it's a tensor
    accuracy = xm.mesh_reduce('accuracy', accuracy, np.mean)
    print(accuracy)
```

**Comment by 8key** [comment]
```
def _mp_fn():
    #some training and valuation code here

    return accuracy

def main():
    accuracy = xmp.spawn(_mp_fn, nprocs=8)
    #some other scripts that are depending on accuracy, but don't need any multiprocessing training
    some_other_function(accuracy)
```

@taylanbil Yea, the purpose is the get some return value and use it in subsequent process, see the update one above. 

Still wondering how to achieve this. Currently I was just writing it to a file and read it back in the main() function.

**Comment by taylanbil** [comment]
`xmp.spawn` just uses `torch.multiprocessing.start_processes` underneath ([here](https://github.com/pytorch/xla/blob/master/torch_xla/distributed/xla_multiprocessing.py#L388)) and it does not return an artifact like accuracy above.

I'd say your options are,

- either use `xm.mesh_reduce`, `xm.all_gather` or similar to sync data between processes, and roll `some_other_function` into `_mp_fn`
- or write to files from subprocesses, and fire up a single process reading the files and parsing/processing them.

**Comment by 8key** [comment]
@taylanbil Good to know, thanks!
